pdns/recursordist/rec-main.cc

```diff
@@ -809,7 +809,13 @@ static void setupNODGlobal()
 
 static void daemonize(Logr::log_t log)
 {
-  if (fork() < 0) {
+  if (auto pid = fork(); pid != 0) {
+    if (pid < 0) {
+      int err = errno;
+      SLOG(g_log << Logger::Critical << "Fork failed: " << stringerror(err) << endl,
+           log->error(Logr::Critical, err, "Fork failed"));
+      exit(1); // NOLINT(concurrency-mt-unsafe
+    }
     exit(0); // NOLINT(concurrency-mt-unsafe
```

Comment:
Just a nit, but unbalanced parenthesis are triggering my OCD :D
```suggestion
      exit(1); // NOLINT(concurrency-mt-unsafe)
    }
    exit(0); // NOLINT(concurrency-mt-unsafe)
```